### ISSUE TITLE
subclassing React.Component prevents property not found errors

### ISSUE THREAD

**Issue by danvk** [facebook/flow#1040]
In subclasses of `React.Component`, calling non-existent methods is not an error.

With Flow 0.18.1, this fails to type check:

```javascript
/** @flow */

class Foo {
  hello() {
    return this.goodbye();
  }
}
```

```
hello.js:5
  5:     return this.goodbye();
                ^^^^^^^^^^^^^^ call of method `goodbye`
  5:     return this.goodbye();
                     ^^^^^^^ property `goodbye`. Property not found in
  3: class Foo {
           ^^^ Foo

Found 1 error
```

If I make `Foo` subclass `React.Component`, though, Flow has no complaints:

```javascript
/** @flow */

var React = require('react');

class Foo extends React.Component {
  hello() {
    return this.goodbye();
  }
}
```

```
No errors!
```

Following https://github.com/facebook/flow/issues/277#issuecomment-152767982, I can fix this by using a more cumbersome type:

```javascript
/** @flow */

var React = require('react');

class Foo extends (React.Component : typeof ReactComponent) {
  hello() {
    return this.goodbye();
  }
}
```

(yields same error as above)

I would expect the second example to produce errors, so that the complex subclass declaration isn't necessary. cc @cloudkite

**Comment by samwgoldman** [comment]
Have you tried either a) adding `.*/node_modules/react/.*` to the `[ignore]` section of your `.flowconfig` or b) using a recent pull from GitHub (at least including 189b109c422fc3fcf8bec97a9598c40ca02add01) that should fix #447?

**Comment by danvk** [comment]
`[ignore]`-ing `.*/node_modules/react/.*` does fix this. Thanks for the tip!
